src/uuid.cr

```diff
@@ -321,6 +325,30 @@ struct UUID
     end
   {% end %}
 
+  # Generates an RFC9562-compatible v7 UUID, allowing the values to be sorted
+  # chronologically (with 1ms precision) by their raw or hexstring
+  # representation.
+  def self.v7(random r : Random = Random::Secure)
+    buffer = uninitialized UInt8[18]
+    value = buffer.to_slice
+
+    # Generate the first 48 bits of the UUID with the current timestamp. We
+    # allocated enough room for a 64-bit timestamp to accommodate the
+    # NetworkEndian.encode call here, but we only need 48 bits of it so we chop
+    # off the first 2 bytes.
+    IO::ByteFormat::NetworkEndian.encode Time.utc.to_unix_ms, value
+    value = value[2..]
+
+    # Fill in the rest with random bytes
+    r.random_bytes(value[6..])
+
+    # Set the version and variant. They are normally in bytes 6 and 8, respectively, but we have 2 superfluous bytes at the front of our buffer that are to be
```

Comment:
**Question** the comment is outdated, right? you already shoved off the first 2 bytes above.

```suggestion
    # Set the version and variant
```

Reply:
Nice catch! I rearranged this method several times before landing on this implementation and this comment was one of the first I wrote.